### PR TITLE
Support for GROMACS 2023

### PR DESCRIPTION
**Pull request by HubLot**
Add patch files for GROMACS 2023 and modify Github Actions

### CODE REVIEW

**Comment by HubLot** [comment]
erf... tests are failing... I don't know why and I don't know why the GROMACS 2023 isn't building.

File: update-colvars-code.sh [hunk 1]
```diff
@@ -558,7 +567,7 @@ then
   echo ""
 
   # Apply patch for Gromacs files
-  patch ${patch_opts} -d ${target} < ${source}/gromacs/gromacs-${GMX_VERSION}.patch
+  # patch ${patch_opts} -d ${target} < ${source}/gromacs/gromacs-${GMX_VERSION}.patch
```
Comment (HanatoK) [inline update-colvars-code.sh:570]: Maybe you forget to apply the GROMACS patch?

Comment (HubLot) [reply update-colvars-code.sh:570]: 😩

Comment (HubLot) [reply update-colvars-code.sh]: thanks!

**Comment by giacomofiorin** [comment]
@HubLot The GROMACS 2023 build job fails because the CMake version from CentOS 7 is too old. `master` now contains the newer container, you should only need to rebase and add `container_name` to the Actions recipe.

**Comment by HubLot** [comment]
Okay, I think the branch is ready now

**Pull request merged**